dsn.go

```diff
@@ -13,21 +13,55 @@
 
 package goalisa
 
-// Config is the deserialization of connect string
+import (
+	"fmt"
+	"net/url"
+	"regexp"
+)
+
+var (
+	reDSN = regexp.MustCompile(`^([a-zA-Z0-9_-]+):([=a-zA-Z0-9_-]+)@([:a-zA-Z0-9/_.-]+)\?([^/]+)$`)
```

Comment:
`[a-zA-Z0-9_-] ` is an equivalent of `\w`, should we use `\w` for better readability?